addons/mail/static/src/discuss/call/common/rtc_service.js

```diff
@@ -327,14 +336,58 @@ export class Rtc extends Record {
             }
             this.call();
         }, 30_000);
+
+        this.busService.subscribe("discuss.rtc/recover", (params) => {
+            this.attemptRecovery(params);
+        });
+
+        this.multiTab.subscribe("discuss.rtc/recover", (params) => {
+            this.attemptRecovery(params);
+        });
+    }
+
+    attemptRecovery({ channel_id, recover_ts }) {
+        if (this.recoveryTimestamp !== recover_ts) {
+            this.recoveryTimestamp = recover_ts;
+            return;
+        }
+        setTimeout(async () => {
+            if (!this.multiTab.isOnMainTab()) {
+                return;
+            }
+            const channel = await this.store.Thread.getOrFetch({
```

Comment:
The logic to pass something with broadcast that is a necessary condition for `sendBeacon() => _bus_send()` is good, but it shouldn't be the same payload.

For example, with current code, you might close a call twice, and due to triggering `_bus_send()` twice on another device, this other device might make a call recovery attempt.

So you should make the payload just different enough to ensure one needs the other one, and not twice broadcast or twice `_bus_send()` for it to succeed.